### DISCUSSION TITLE
Attaching RS485 board to Wifi-Dongle

### DISCUSSION THREAD

**Discussion by odroegehorn** [syssi/esphome-soyosource-gtn-virtual-meter#96] [General]
Hi everybody,
I just want to document, what I did with the help of the software-developer #syssi:

I flashed the stock-wifi dongle with the new 8266-wifi-dongle firmware with the "example.yaml" file for it.
This worked already. Beyond that, I added the web-server support to it (as shown in "display-example.yaml").

Last I have asked, if there are free GPIOs on the 8266, and yes, there are.
In my case, I have soldered an RS485-to-TTL module (same as shown on the first page) directly under the wifi-dongle.
+3.3V and GND are from the dongle-board (see dongle-doc) and RX/TX are going to the 8266.
I used GPIO5 (as it is free) and GPIO14 (as it was presoldered), easiest way to solder for me. Originally I planned for GPIO5 & 16, but the 16 PIN didn't got strong soldering for the cable, therefore I switched to GPIO14.
You can see the results in the pictures.
All fit's "more or less" into the original wifi-dongle housing. You just have to make a little space for the RS485 board, and widen the USB-cut-out a little bit, as the wifi-dongle board comes a little bit up.

The rest is straight forward. I integrated the RS485 like in the "display-example.yaml" file, so that we have WIFI-TTL and RS485 on two uarts, like in the file. Only difference: it's not the WeMos but the wifi-dongle.

Hope that helps anybody.

![IMG_4787](https://user-images.githubusercontent.com/75804802/211036594-a4cadf3b-284d-4629-a715-3842cbad7264.JPG)
![IMG_4783](https://user-images.githubusercontent.com/75804802/211036600-b1901f13-d182-4cd6-8925-e63802966d64.JPG)
![IMG_4784](https://user-images.githubusercontent.com/75804802/211036608-f66c65df-c926-4212-9e48-d0b1c27347df.JPG)
![IMG_4785](https://user-images.githubusercontent.com/75804802/211036613-0dad4646-a487-498b-ba73-4849fab0c029.JPG)
![IMG_4786](https://user-images.githubusercontent.com/75804802/211036615-44f8e96b-0aa1-4e4a-9c56-22924598a779.JPG)

**Comment by syssi** [comment]
Thanks a lot for your summary! Feel free to create new issues if you miss something or if you see room for improvements.
